packages/analytics-browser/src/config.ts

```diff
@@ -75,6 +78,7 @@ export class BrowserConfig extends Config implements IBrowserConfig {
     },
     public transport: 'fetch' | 'xhr' | 'beacon' = 'fetch',
     public useBatch: boolean = false,
+    public diagnosticProvider: IDiagnostic | DiagnosticOptions = new Diagnostic(),
```

Comment:
This make the usage elsewhere a little crufty with type casts. Can you change this to only accept a provider (not options) similar to transportProvider and storage. 
e.g.
https://github.com/amplitude/Amplitude-TypeScript/pull/568/files#diff-379abfc2a0d72883f73638e5ee6b8a9e0fb89793f5cebc6751a0716731bfa1d7R99

Reply:
verify the payload 

```
{
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        api_key: '',
        omni_metrics: [
          {
          metadata_type: DIAGNOSTIC_METADATA_TYPE,
          library: 'amplitude-ts',
          accounting_time_min: 12345,
          response_code: 400,
          trigger: 'test message 0',
          action: 'drop events',
          event_count: 5,
        }
      ],
    })
}
```

Reply:
Removed DiagnosticOptions in https://github.com/amplitude/Amplitude-TypeScript/pull/568/commits/3ad286e0a5203570c62c5cd6099e64f3f2a905cf